core/tests/puppeteer-acceptance-tests/user-utilities/exploration-editor-utils.ts

```diff
@@ -262,18 +269,38 @@ export class ExplorationEditor extends BaseUser {
    * Deletes the previous written title and updates the new title.
    */
   async updateTitleTo(title: string): Promise<void> {
+    await this.page.waitForSelector(explorationTitleSelector);
+    const titleInput = await this.page.$(explorationTitleSelector);
+    const oldTitle = await this.page.evaluate(input => input.value, titleInput);
     await this.clearAllTextFrom(addTitleBar);
     await this.type(addTitleBar, title);
     await this.page.keyboard.press('Tab');
+    // Auto save pop up bar is visible only when current input is different from
+    // old input.
+    if (oldTitle !== title) {
+      if (this.isViewportAtMobileWidth()) {
+        // Navbar text is hidden in mobile view port due to less screen so there is no visible
+        // change in the UI (specially Auto save pop up bar) after we update the input bar.
+        // Hence we need to explicitly wait for 2 seconds.
+        await this.page.waitForTimeout(2000);
```

Comment:
Are there any network requests after updating the input bar? If so, maybe try `page.waitForNetworkIdle()` instead of `page.waitForTimeout(2000)`? (Documentation for the method is [here](https://pptr.dev/api/puppeteer.page.waitfornetworkidle))

Reply:
No network request. Tried it also. Thanks for this.

Reply:
Waiting for a fixed timeout doesn't guarantee that the autosave has completed. It's a bit of a guess and could lead to flaky tests if the autosave takes longer than expected. Maybe we can confirm the same via some other way?

Reply:
[Have to use 2 seconds.webm](https://github.com/oppia/oppia/assets/136263179/f4cf8afe-b47b-43db-9dec-e8f758aede89)
PTAL. No UI change, no network request. I've attempted over 30 times. It's not exceeding 1.6 seconds. For ease, we've set it at 2 seconds (to avoid flakiness). If you have any ideas, feel free to suggest. Thanks :)

Reply:
No waiting in acceptance tests. This is an absolute rule. If there is any issue with something not being clear in the UI then please fix the UI, because if the acceptance test does not know when something finishes then an end user will not know either.

Reply:
@rahat2134 Oh, okay. Also GitHub won't show me the video you linked because of the video format. Maybe try changing the video format to .mp4?

Is there enough space on the screen to have the auto save pop up bar show up? Or if that's not the case, maybe we can display a more mobile-friendly auto save pop up bar? Either way, like @seanlip mentioned, I think fixing the UI so that there is some visible UI state change would be the way to go.

Reply:
@rahat2134 Okay. I'll approve this PR.

Also, I really can't see the video because the video format is not supported, at least on my machine. It might have something to do with my browser or my OS.

![cant-see-video](https://github.com/oppia/oppia/assets/117241994/8d0362d2-36df-412d-9dbc-f06d76b8c189)

Reply:
@seanlip @Akhilesh-max is the video visible in your machines?

Reply:
The video is visible but "weird". I cannot scrub it or see the full length of the video timeline. I suggest using a different format too if possible.

Reply:
Yes, this video format took a while to load and seek the video.

Reply:
Is it worth using a different format to upload the video, then? Other contributors' videos don't seem to have this problem...

Reply:
Sure, I will start using a different video format from now on. Thank you :)